### PR TITLE
py-zbar: new port

### PR DESCRIPTION
**Pull request by catap**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.5.2 20G95 x86_64
Xcode 12.5.1 12E507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-zbar/Portfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+platforms           darwin
+supported_archs     noarch
+license             MIT
+
+maintainers         {@catap korins.ky:kirill} openmaintainer
+
+description         Read one-dimensional barcodes and QR codes from Python 2 and 3.
+long_description    ${description}
+
+homepage            https://github.com/NaturalHistoryMuseum/pyzbar
```
Comment (reneeotten) [inline python/py-zbar/Portfile:21]: homepage is set automatically when using the github PG

File: python/py-zbar/Portfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+checksums           rmd160  e863508491e4237c91d5c3ea8c1d79f601a9a630 \
+                    sha256  ce49f08ad1ede72cf8d9df43de83be7cfda54df9da9bd77cc027cc3a58c05036 \
+                    size    41545
+
+python.versions     37 38 39
+
+python.link_binaries no
```
Comment (reneeotten) [inline python/py-zbar/Portfile:30]: why is this necessary? If you do provide the port for different Python versions then you'd likely want to add the -X.Y version, or...?

File: python/py-zbar/Portfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+license             MIT
+
+maintainers         {@catap korins.ky:kirill} openmaintainer
+
+description         Read one-dimensional barcodes and QR codes from Python 2 and 3.
+long_description    ${description}
+
+homepage            https://github.com/NaturalHistoryMuseum/pyzbar
+
+
```
Comment (reneeotten) [inline python/py-zbar/Portfile:23]: you probably didn't intend to have two newlines here?

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-zbar/Portfile (hunk 2 above)
Comment (catap) [reply python/py-zbar/Portfile:30]: because this port adds two binary:
```
/opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/read_zbar
/opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/read_zbar.py
```
and one of them from port's scripts folder.

If it ok, I can remove this line

Comment (reneeotten) [reply python/py-zbar/Portfile:27]: ah, I see. I would just add a patchfile to remove the `scripts` line in the setup.py and then remove the line in question here. I think that will give you the expected end results.

Comment (catap) [reply python/py-zbar/Portfile:27]: @reneeotten I don't think this is the right way because `scripts` version seems the right one. Let me share here:
```
% cat /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/read_zbar
# EASY-INSTALL-ENTRY-SCRIPT: 'pyzbar==0.1.8','console_scripts','read_zbar'
import re
import sys

# for compatibility with easy_install; see #2198
__requires__ = 'pyzbar==0.1.8'

try:
    from importlib.metadata import distribution
except ImportError:
    try:
        from importlib_metadata import distribution
    except ImportError:
        from pkg_resources import load_entry_point


def importlib_load_entry_point(spec, group, name):
    dist_name, _, _ = spec.partition('==')
    matches = (
        entry_point
        for entry_point in distribution(dist_name).entry_points
        if entry_point.group == group and entry_point.name == name
    )
    return next(matches).load()


globals().setdefault('load_entry_point', importlib_load_entry_point)


if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(load_entry_point('pyzbar==0.1.8', 'console_scripts', 'read_zbar')())
```
versus
```
% cat /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/read_zbar.py
#!/opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9
from __future__ import print_function

import argparse
import sys

import pyzbar
from pyzbar.pyzbar import decode


def main(args=None):
    if args is None:
        args = sys.argv[1:]

    parser = argparse.ArgumentParser(
        description='Reads barcodes in images, using the zbar library'
    )
    parser.add_argument('image', nargs='+')
    parser.add_argument(
        '-v', '--version', action='version',
        version='%(prog)s ' + pyzbar.__version__
    )
    args = parser.parse_args(args)

    from PIL import Image

    for image in args.image:
        for barcode in decode(Image.open(image)):
            print(barcode.data)


if __name__ == '__main__':
    main()
```

I can rename `read_zbar.py` to `read_zbar` at the last place, but maybe I have cleaner way?

Comment (reneeotten) [reply python/py-zbar/Portfile:27]: did you try what I suggested? Because I did that myself before adding the previous comment and it seems to work fine for me...

Comment (catap) [reply python/py-zbar/Portfile:27]: @reneeotten sorry, I read your commit without the first coffee and I read it as remove `scripts` directory from sources.

Comment (reneeotten) [reply python/py-zbar/Portfile]: now you removed the part with the entry_points / console_script, that was not what I meant. A few lines above is a "scripts" entry, I think that's the one you want to remove. I suppose at least you want to have the read_zbar script available.

Comment (catap) [reply python/py-zbar/Portfile]: @reneeotten I want to have `read_zbar.py` from authors which can be used to use this library to read barcode data at image.

Comment (reneeotten) [reply python/py-zbar/Portfile]: I don't think that is what you want ;) If you'd do as I suggested this is the end result regarding installed files:
```
 % port contents py39-zbar
Port py39-zbar contains:
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/read_zbar
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar-0.1.8-py3.9.egg-info/PKG-INFO
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar-0.1.8-py3.9.egg-info/SOURCES.txt
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar-0.1.8-py3.9.egg-info/dependency_links.txt
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar-0.1.8-py3.9.egg-info/entry_points.txt
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar-0.1.8-py3.9.egg-info/requires.txt
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar-0.1.8-py3.9.egg-info/top_level.txt
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__init__.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__pycache__/__init__.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__pycache__/locations.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__pycache__/pyzbar.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__pycache__/pyzbar_error.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__pycache__/wrapper.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/__pycache__/zbar_library.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/locations.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/pyzbar.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/pyzbar_error.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/scripts/__init__.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/scripts/__pycache__/__init__.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/scripts/__pycache__/read_zbar.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/scripts/read_zbar.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/__init__.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/__pycache__/__init__.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/__pycache__/test_locations.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/__pycache__/test_pyzbar.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/__pycache__/test_read_zbar.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/__pycache__/test_zbar_library.cpython-39.pyc
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/test_locations.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/test_pyzbar.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/test_read_zbar.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/tests/test_zbar_library.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/wrapper.py
  /opt/local/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pyzbar/zbar_library.py
  /opt/local/bin/read_zbar-3.9
```

so you'll have `read_zbar-3.9` that will do what you want and refers to the file `scripts/read_zbar.py`: 
```
% read_zbar-3.9
usage: read_zbar-3.9 [-h] [-v] image [image ...]
read_zbar-3.9: error: the following arguments are required: image
```

To me that appears to be what you want. If however, you want *both* the `read_zbar-3.9` and `read_zbar.py-3.9` in the `${prefix}/bin` directory then you don't do any patches and just leave it as upstream apparently does it. In any case, with the Python ports where you allow multiple versions I think you do need to link the binaries. So just pick what you preferred option is and let's go with that - I don't have the time to keep going back-and-forth about this today.

Comment (catap) [reply python/py-zbar/Portfile]: @reneeotten this is that I wont.

I suggest to keep it this way, and if someone needed something different (really doubt that) => let fix it.

Comment (reneeotten) [reply python/py-zbar/Portfile]: > @reneeotten I want to have `read_zbar.py` from authors which can be used to use this library to read barcode data at image.

You will always get that anyway (as in `/opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/read_zbar.py`). Then let's just do what upstream wants and install both `read_zbar` and `read_zbar.py` ; so just remove the patch and we should be good to go.

Comment (catap) [reply python/py-zbar/Portfile]: @reneeotten I agree with you that let keep as upstream wants.

**Review by reneeotten — APPROVED** [review]
